docs/error-codes.md

```diff
@@ -1,135 +1,73 @@
-# IL Linker Error Codes
-# IL Linker Warning Codes 
+# ILLinker Errors and Warnings
 
-### IL2001
-```
-Type 'type' has no fields to preserve
-```
+Every ILLinker error and warning has assign unique error code for easier
+identification. The known codes are in the range 1000 to 6000. A custom
+steps should avoid using this range not to collide with ILLinker error and
+warning codes.
 
-The XML descriptor preserves fields on type 'type', but this type has no fields.
+## Error Codes
 
----
+#### IL1001: To be added
 
-### IL2002
-```
-Type 'type' has no methods to preserve
-```
 
-The XML descriptor preserves methods on type 'type', but this type has no methods.
+## Warning Codes
 
----
+#### IL2001: Type 'type' has no fields to preserve
 
-### IL2003
-```
-Could not resolve 'assembly' assembly dependency specified in a `PreserveDependency` attribute that targets method 'method'
-```
+The XML descriptor preserves fields on type 'type', but this type has no fields.
 
-The assembly 'assembly' in `PreserveDependency` attribute could not be resolved.
+#### IL2002: Type 'type' has no methods to preserve
+
+The XML descriptor preserves methods on type 'type', but this type has no methods.
 
----
+#### IL2003: Could not resolve 'assembly' assembly dependency specified in a 'PreserveDependency' attribute that targets method 'method'
```

Comment:
Just a preference - I personally like the style that puts the warning text on another line, to avoid long headings like these. That's also more similar to what the roslyn docs do: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/cs0034

Reply:
I went a bit back and forth on this and the recent GH style change made it worse but as we don't have a lot of text there using ILXXXX at heading only makes the page look empty and some of the details are very similar to the error message. Btw, the view and preview in VS Code, for example, looks much better.